python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -338,16 +410,45 @@ cdef class Column:
         Parameters
         ----------
         obj : object
-            Must implement the `__array_interface__` protocol.
+            Must implement the ``__array_interface__`` protocol.
+
+        Returns
+        -------
+        Column
+            A Column containing the data from the array interface.
 
         Raises
         ------
+        ImportError
+            If NumPy is not installed.
+        TypeError
+            If the object does not implement ``__array_interface__``
+        ValueError
+            If the host array is not 1D or 2D, or is not C-contiguous.
+            If the number of rows exceeds size_type limit.
         NotImplementedError
-            This method is not yet implemented.
+            If the object has a mask.
         """
-        raise NotImplementedError(
-            "Converting to a pylibcudf Column is not yet implemented."
-        )
+        try:
+            import numpy as np
+            iface = obj.__array_interface__
+        except ImportError:
+            raise ImportError("NumPy must be installed to use this method on 2D data")
+        except AttributeError:
+            raise TypeError("Object does not implement __array_interface__")
+
+        arr = np.asarray(obj)
```

Comment:
Just noting that if `obj` supports the buffer protocol or `iface["data"]` indicates supporting the buffer protocol I think we can just use `memoryview` directly instead of going though numpy e.g. `device_buffer = DeviceBuffer.to_device(memoryview(obj))`


> This attribute can also be an object exposing the [buffer interface](https://docs.python.org/3/c-api/buffer.html#bufferobjects) which will be used to share the data. If this key is not present (or returns None), then memory sharing will be done through the buffer interface of the object itself. In this case, the offset key can be used to indicate the start of the buffer. A reference to the object exposing the array interface must be stored by the new object if the memory area is to be secured.

Reply:
I can check if the object supports the buffer protocol with [`PyObject_CheckBuffer`](https://docs.python.org/3/c-api/buffer.html#c.PyObject_CheckBuffer).

Also using `memoryview(...)` is probably fine too (but it's potentially not `O(1)`). Maybe only call `memoryview` if `PyObject_GetBuffer` returns 0.
```python
def _supports_buffer_protocol():
    try:
        return PyObject_CheckBuffer(obj) or memoryview(obj)
    except TypeError:
        return False
```
CC @vyasr

Reply:
I don't have a scenario in mind. I know it calls `__buffer__` and we don't know whether its O(1) or not.

Reply:
I don't think `DeviceBuffer.to_device(memoryview(obj))` will work because we need a reliable way of viewing the data as unsigned char. I think the best thing to do here is to just use numpy.

Reply:
Ok if something like `memoryview(obj).cast("c")` doesn't work I suppose numpy is a good catch-all. Would be nice to investigate in the future if we do not need to rely on numpy for this

Reply:
I didn't get to comment earlier here, but it seems like we found a solution avoiding numpy altogether by just checking the array interface properties which seems ideal.